### ISSUE TITLE
Implement a learning rate scheduler

### ISSUE THREAD

**Issue by RyanMeline** [RyanMeline/MNIST_ANN#3]
- [ ] Reduce the learning rate on percent plateau 

- [x] Step decay 
ex: epochs 1-5 have .01 learning rate, 6-10 have .001, 11-15 have .0001
if(i % 5 == 0) learning_rate *= 0.1f;

- [ ] Exponential decay
lr = initial_lr * e^(-decay_rate * epoch)

- [x] Cosine curve
lr = min_lr + 0.5*(max_lr - min_lr) * (1 + cos(epoch/max_epochs * π))
